x-pack/plugins/canvas/public/lib/keymap.js

```diff
@@ -38,36 +38,81 @@ const getCtrlShortcuts = shortcuts => {
   };
 };
 
-const refreshShortcut = getAltShortcuts('r');
-const previousPageShortcut = getAltShortcuts('[');
-const nextPageShortcut = getAltShortcuts(']');
+const refreshShortcut = { ...getAltShortcuts('r'), help: 'Refresh workpad' };
+const previousPageShortcut = { ...getAltShortcuts('['), help: 'Go to previous page' };
+const nextPageShortcut = { ...getAltShortcuts(']'), help: 'Go to next page' };
 
 export const keymap = {
+  ELEMENT: {
+    displayName: 'Element controls',
+    COPY: { ...getCtrlShortcuts('c'), help: 'Copy elements' },
+    CLONE: { ...getCtrlShortcuts('d'), help: 'Clone elements' },
+    CUT: { ...getCtrlShortcuts('x'), help: 'Cut elements' },
+    PASTE: { ...getCtrlShortcuts('v'), help: 'Paste' },
+    DELETE: {
+      osx: ['backspace'],
+      windows: ['del', 'backspace'],
+      linux: ['del', 'backspace'],
+      other: ['del', 'backspace'],
+      help: 'Delete elements',
+    },
+    BRING_FORWARD: {
+      ...getCtrlShortcuts('up'),
+      help: 'Send element forward one layer',
+    },
+    SEND_BACKWARD: {
+      ...getCtrlShortcuts('down'),
+      help: 'Send element back one layer',
+    },
+    BRING_TO_FRONT: {
+      ...getCtrlShortcuts('shift+up'),
+      help: 'Send element to front',
+    },
+    SEND_TO_BACK: {
+      ...getCtrlShortcuts('shift+down'),
+      help: 'Send element to back',
+    },
+    GROUP: {
+      osx: ['g'],
+      windows: ['g'],
+      linux: ['g'],
+      other: ['g'],
+      help: 'Group elements',
+    },
+    UNGROUP: {
+      osx: ['u'],
+      windows: ['u'],
+      linux: ['u'],
+      other: ['u'],
```

Comment:
If all platforms have the same key, would a single line of `other` cover it as a fallback, or is there some other defaulting? I'm not saying enlisting it is not nice, eg. for code documentation purpose.

Reply:
It looks like react-shortcuts handles both cases. Both lowercase and uppercase keys (i.e with caps lock on) trigger the shortcuts.

Reply:
Oh I needed the objects to be a specific shape. A single line would've worked too, but I wanted the shortcut to with the shape 
```javascript
{
help: 'description',
osx: ['shortcut'],
windows: ['shortcut'],
linux: ['shortcut'],
other: ['shortcut'],
}
``` 
because I needed the shortcut to be an object that I could add a `help` attribute to that is compatible with what `react-shortcuts` expects as a valid keymap.